src/init.cpp

```diff
@@ -537,6 +536,21 @@ bool InitSanityCheck(void)
     return true;
 }
 
+bool AppInitServers(boost::thread_group& threadGroup)
+{
+    RPCServer::OnStopped(&OnRPCStopped);
+    RPCServer::OnPreCommand(&OnRPCPreCommand);
+    if (!StartHTTPServer(threadGroup))
```

Comment:
I think there's a complicated race condition here.  StartHTTPServer() will bind the HTTP server and start listening, but if requests come in before the RPC/REST servers have called RegisterHTTPHandler(), their requests will result in 404s.  Moreover, if the request comes in before eventBase is set in StartHTTPServer(), HTTPRequest::WriteReply() will attempt to queue an HTTPEvent with an invalid event_base, which results in the server dropping the request on the floor and not replying at all.  (with an ugly libevent warning appearing on the console)  You can reproduce this by adding a MilliSleep(5000) to the end of StartHTTPServer before it assigns eventBase.  With this in place, regression tests will fail.  (It's pretty sporadic without the sleep)  You can catch the null eventBase by adding an assertion to HTTPRequest::WriteReply(), where you should see nStatus being set to 404.  Hope this is helpful!